File: Data Structures & Algorithms/max-area-of-island/submission-0.java
class Solution {

    public int dfs(int[][] grid, boolean[][] vis, int x, int y) {
        if (x < 0 || y < 0 || x >= grid.length || y >= grid[0].length || 
            vis[x][y] || grid[x][y] == 0) {
            return 0;
        }

        vis[x][y] = true;

        int area = 1;
        area += dfs(grid, vis, x + 1, y);
        area += dfs(grid, vis, x - 1, y);
        area += dfs(grid, vis, x, y + 1);
        area += dfs(grid, vis, x, y - 1);

        return area;
    }

    public int maxAreaOfIsland(int[][] grid) {
        int n = grid.length;
        int m = grid[0].length;
        boolean[][] vis = new boolean[n][m];
        int res = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (!vis[i][j] && grid[i][j] == 1) {
                    int t = dfs(grid, vis, i, j);
                    res = Math.max(res, t);
                }
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n=len(prices)
        l=[0 for _ in range(n)]
        t=prices[0]
        l[0]=prices[0]
        for i in range(n):
            t=min(t,prices[i])
            l[i]=t
        i=n-1
        res=0
        print(l)
        while i>=1:
            res=max(res,prices[i]-l[i-1])
            i-=1
        return res